### PR TITLE
Add method VectorDiff::apply

### PR DESCRIPTION
**Pull request by progval**

### CODE REVIEW

**Comment by jplatte** [comment]
Hey, thanks for the PR! Did you consider exposing this as `ObservableVector::apply_diff` instead? I think `vector.apply_diff(diff)` is probably easier to read than `diff.apply(&mut vector)`.

**Comment by progval** [comment]
Hmm, how about a `trait PatchableVector { fn apply_diff(&mut self, diff: VectorDiff); }`, implemented by `imbl::Vector` (for my use case), `ObservableVector` (for yours), and possibly others like `Vec`?

File: eyeball-im/src/vector.rs [hunk 1]
```diff
@@ -413,8 +413,33 @@ impl<T: Clone> VectorDiff<T> {
...
+            VectorDiff::Insert { index, value } => vec.insert(index, value),
+            VectorDiff::Set { index, value } => { vec.set(index, value); },
+            VectorDiff::Remove { index } => { vec.remove(index); },
+            VectorDiff::Truncate { length } => vec.truncate(length),
+            VectorDiff::Reset { values } => { *vec = values },
+        }
+    }
 }
 
+
```
Comment (jplatte) [inline eyeball-im/src/vector.rs:442]:
```suggestion
```

**Comment by jplatte** [comment]
Hm, I think formatting is off here, but we don't have a CI job to check that. I guess I'll merge and take care of the remaining work separately.

**Comment by jplatte** [comment]
Released in `eyeball-im 0.4.2`.

**Pull request merged**